docker/tests/factories/http_response_factory.py

```
@@ -3,94 +3,122 @@
-    Returns proper test doubles, not MagicMock with property assignment.
-    Each method returns a fully-configured MockHttpResponse instance.
-    """
+    """Factory for creating pre-configured HTTP response mocks."""
 
     @staticmethod
     def success(
-        content: str = "<html><body><h1>Test Page</h1></body></html>",
-        content_type: str = "text/html; charset=utf-8",
-        status_code: int = 200,
+        content: bytes = b"<html><body><p>Test content</p></body></html>",
+        headers: dict = None,
```

Ruff (0.13.2)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)